### DISCUSSION TITLE
Is Quarto Compatible with Windows Task Scheduler?

### DISCUSSION THREAD

**Discussion by mm225022** [quarto-dev/quarto-cli#5230] [Q&A] [answered]
I have been able to schedule RMarkdown reports to be generated automatically via the Windows Task Scheduler.  [John Mackintosh's post](https://johnmackintosh.net/blog/2021-11-01-scheduling/) was quite helpful for troubleshooting.
However, I have been unable to reproduce the automation via Task Scheduler for Quarto.
I tried, unsuccessfully, to schedule a Quarto report with Task Scheduler.
I took the same report, converted it to an RMarkdown report, ran the same R script (except this time calling render as opposed to quarto_render) and it ran via Window's Task Scheduler.
Is there anything else that I need to do to automate a Quarto report via Window's Task Scheduler?  Is it possible to to this?  Is there any other preferred way?  
Thanks

**Comment by cderv** [comment, marked as answer]
I would say there is nothing specific about Quarto that would prevent running from task scheduler using any of the approach in the post. 

Quarto is CLI tool first, so I would use a .bat (cmd) or .ps1 (powershell) script that would call `quarto render` for your document.
If you want to go the R way, then you can use the **quarto** R package indeed. 

Only thing you need to take care of is be sure that Windows can find `quarto` binary from the scheduled tasks. I am thinking that if you are using Quarto only from RStudio for example, the CLI tool won't be in PATH, and only accessible from within RStudio IDE. 
If you install Quarto with installer (https://quarto.org/docs/get-started/) then it should be put in PATH. 

With R package you can also set the `QUARTO_PATH` environment variable to point to the Quarto binary if needed. 

I think this is a configuration issue you are encountering. Windows task scheduler should be working with any program - unless I am unaware of some limitation. 

If you have some error message to share, I can try help further

**Comment by mm225022** [comment]
Thank you @cderv, that fixed it.
In my original file, I had the path to Pandoc - `Sys.setenv(RSTUDIO_PANDOC = 'C:/Program Files/RStudio/resources/app/bin/quarto/bin/tools')`.  Based on what you wrote, I added the Path to the Quarto binary - `Sys.setenv(QUARTO_PATH = 'C:/Program Files/RStudio/resources/app/bin/quarto/bin/quarto.exe')`.

It now works.

Thanks!

**Reply by cderv** [reply]
Glad it works ! Path to PANDOC shouldn't be needed if you are using Quarto. I wouldn't set it unless you are using `rmarkdown::render()` directly in your process too.

**Comment by johnmackintosh** [comment]
Just in case this helps anyone else- 

I was testing setting a job up with quarto and could not get the task to run with 

```
Sys.setenv(QUARTO_PATH = 'C:/Program Files/RStudio/resources/app/bin/quarto/bin/quarto.exe')
```

However, removing the `.exe` worked and the job has run via scheduler and .bat file
```
Sys.setenv(QUARTO_PATH = 'C:/Program Files/RStudio/resources/app/bin/quarto/bin/quarto')

```

(I don't have the CLI installed so running this via R script and the `{quarto}` package)

**Reply by cderv** [reply]
Which version of Quarto are you running ? Maybe this is an old one when `.exe` file was not available yet.

**Reply by johnmackintosh** [reply]
I have Quarto version 1.2.269
R version 4.2.2
RStudio 2022.12.0

**Reply by cderv** [reply]
> I have Quarto version 1.2.269

Quarto 1.3 was released today. Can you update to this one ?

**Reply by johnmackintosh** [reply]
I can't update quarto, but having updated RStudio (which appears to be quarto version 1.2.475?) and now I do need to specify the ```.exe``` extension. Thanks
